### PR TITLE
Fix race condition when installing 2+ editable non-VCS pkgs at once

### PR DESCRIPTION
**Pull request by christek91**
This regression was recently introduced, and only affects non-vcs
packages. The effect of the race is that installations of multiple
editable non-VCS sourced packages at once may cause some of them to be
un-importable.

The fix is to make editable package installs Blocking just like VCS
installs are.

Thank you for contributing to Pipenv!

### The issue

This fixes a race condition that is hit when installing multiple local editable packages at once. There was a regression because `editable` installations used to be blocking, but a change in the recent release changed it so that only VCS installations are blocking. 

While most VCS installations are probably `editable, not all `editable` installations are VCS ones. So that change inadvertently brought back https://github.com/pypa/pipenv/issues/919. 
 
I've also created a new Issue for the new regression as the PR template suggests. It is: https://github.com/pypa/pipenv/issues/3236

### The fix

My fix makes sure that all `editable` installations are blocking, regardless of if they are VCS installs or not. 

The test I wrote for this is a little janky to setup since I couldn't use git to install the test packages since VCS installations don't hit the race. I put a comment in the code explaining the context.

### The checklist

* [x] Associated issue
* [x] A news fragment in the `news/` directory to describe this fix with the extension `.bugfix`, `.feature`, `.behavior`, `.doc`. `.vendor`. or `.trivial` (this will appear in the release changelog). Use semantic line breaks and name the file after the issue number or the PR #.

### CODE REVIEW

File: tests/integration/test_install_twists.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import pytest
 
 from flaky import flaky
+import delegator
```
Comment (frostming) [inline tests/integration/test_install_twists.py]: I guess this line can be removed?

File: tests/integration/test_install_twists.py [hunk 2]
```diff
@@ -333,3 +334,55 @@ def test_install_local_uri_special_character(PipenvInstance, testsroot):
         c = p.pipenv("install")
         assert c.return_code == 0
         assert "six" in p.lockfile["default"]
+
+
+@pytest.mark.files
+@pytest.mark.needs_internet
```
Comment (frostming) [inline tests/integration/test_install_twists.py]: Does it really need internet? You are using the mocked pypi test server in this case.

File: tests/integration/test_install_twists.py [hunk 3]
```diff
@@ -333,3 +334,55 @@ def test_install_local_uri_special_character(PipenvInstance, testsroot):
...
+    once, and which causes some of them to not be importable.
+
+    This issue had been fixed for VCS packages already, but not local 'editable' packages.
+
+    So this test locally installs packages from tarballs that have already been committed in
+    the local `pypi` dir to avoid using VCS packages.
+    """
+    pkgs = {
+        "requests-2.19.1": "requests/requests-2.19.1.tar.gz",
+        "flask-0.12.2": "flask/flask-0.12.2.tar.gz",
```
Comment (frostming) [inline tests/integration/test_install_twists.py]:
```suggestion
        "flask-0.12.2": "flask/Flask-0.12.2.tar.gz",
```

File: tests/integration/test_install_twists.py [hunk 4]
```diff
@@ -333,3 +334,55 @@ def test_install_local_uri_special_character(PipenvInstance, testsroot):
...
+        "six-1.11.0": "six/six-1.11.0.tar.gz",
+        "jinja2-2.10": "jinja2/jinja2-2.10.tar.gz",
```
Comment (frostming) [inline tests/integration/test_install_twists.py]:
```suggestion
        "jinja2-2.10": "jinja2/Jinja2-2.10.tar.gz",
```

File: tests/integration/test_install_twists.py (hunk 1 above)
Comment (christek91) [reply tests/integration/test_install_twists.py]: Ah, good point. That was left over from an earlier iteration. I'll fix that.

File: tests/integration/test_install_twists.py (hunk 2 above)
Comment (christek91) [reply tests/integration/test_install_twists.py]: No, you're right it does not in the final iteration. Will fix that.

File: tests/integration/test_install_twists.py (hunk 3 above)
Comment (christek91) [reply tests/integration/test_install_twists.py]: Thanks, updated in new commit!

File: tests/integration/test_install_twists.py (hunk 4 above)
Comment (christek91) [reply tests/integration/test_install_twists.py]: Thanks, updated in new commit!

File: tests/integration/test_install_twists.py [hunk 5]
```diff
@@ -333,3 +333,55 @@ def test_install_local_uri_special_character(PipenvInstance, testsroot):
...
+
+        import tarfile
+
+        with tarfile.open(source_path, "r:gz") as tgz:
+            tgz.extractall(path=tmpdir.strpath)
+
+        pipfile_string += "'{0}' = {{path = '{1}', editable = true}}\n".format(pkg_name, unzip_path)
+
+    with PipenvInstance(pypi=pypi, chdir=True) as p:
+        print(pipfile_string)
```
Comment (frostming) [inline tests/integration/test_install_twists.py]: Please remove the debugging prints.

**Comment by techalchemy** [comment]
once the debug line is gone im good to merge this, i didn't think about adding editable as a conditional there but i'm glad you were able to find that.  Hopefully the code cleanup efforts will make contributing easier

File: tests/integration/test_install_twists.py (hunk 5 above)
Comment (christek91) [reply tests/integration/test_install_twists.py]: 👍 will do!

**Pull request merged**